src/dask_awkward/lib/core.py

```diff
@@ -959,15 +960,33 @@ def __getitem__(self, where: Any) -> AwkwardDaskCollection:
 
     def _call_behavior_method(self, method_name: str, *args: Any, **kwargs: Any) -> Any:
         """Call a behavior method for an awkward array.
-        Please note that the raw dask_awkward arguments are forwarded to the
-        awkward Mixin class, and the user must deal with when to make calls
-        to map_partitions.
+        If the function signature has __dunder__ parameters it is assumed that the
+        user wants to do the map_partitions dispatch themselves and the _meta's
+        behavior is called.
+        If there are no __dunder__ parameters in the function call then the function
+        is wrapped in map_partitions automatically.
         """
         if hasattr(self._meta, method_name):
-            array_context = kwargs.pop("__dask_array__", self)
-            return getattr(self._meta, method_name)(
-                *args, __dask_array__=array_context, **kwargs
+            themethod = getattr(self._meta, method_name)
+            thesig = inspect.signature(themethod)
+            ndunder = sum(
+                [
+                    1
+                    if len(param) > 4
+                    and param.startswith("__")
+                    and param.endswith("__")
+                    else 0
+                    for param in thesig.parameters
+                ]
+            )
```

Comment:
I'm not sure about checking for `__`. Unless we know that we're going to want to declare a set of parameters, instead I'd prefer an explicit test for `_dask_array_` (using single underscores, as dunder is usually reserved for Python protocols).